index.js

```diff
@@ -70,7 +82,28 @@ class Main {
     metrics.currentBlock.set(this.worker.lastConfirmedBlock);
     metrics.requestsCounter.inc(this.worker.getNewRequestsCount());
     metrics.requestsResponseTime.observe(new Date() - this.worker.lastRequestStartTime);
-    metrics.lastExportedBlock.set(this.worker.lastExportedBlock);
+    if (BLOCKCHAIN === 'eth') {
+      metrics.lastExportedBlock.set(this.taskManager.lastExportedBlock);
+    } else {
+      metrics.lastExportedBlock.set(this.worker.lastExportedBlock);
+    }
+  }
+
+  async waitOnStoreEvents() {
+    const buffer = this.taskManager.retrieveCompleted();
```

Comment:
In your current implementation some logic which is not native for the TaskManager has been put inside it (last exported block, primary keys). It is better to have more separation of concern. For example setting primary keys as per the logic of each blockchain is not native to it, ideally the TaskManager shouldn't even know which blockchain it is working on. Decorating the result data is something which you can do here.